src/pods/generate/GenerateUtil.ts

```diff
@@ -0,0 +1,24 @@
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import type { ResourceStore } from '../../storage/ResourceStore';
+import type { Agent } from '../agent/Agent';
+import type { ResourcesGenerator } from './ResourcesGenerator';
+
+/**
+ * Generates resources with the given generator and adds them to the given store.
+ * @param identifier - Identifier of the pod.
+ * @param agent - Agent from which the pod is being created.
+ * @param generator - Generator to be used.
+ * @param store - Store to be updated.
+ *
+ * @returns The amount of resources that were added.
+ */
+export async function addGeneratedResources(identifier: ResourceIdentifier, agent: Agent,
+  generator: ResourcesGenerator, store: ResourceStore): Promise<number> {
+  const resources = generator.generate(identifier, agent);
+  let count = 0;
+  for await (const { identifier: resourceId, representation } of resources) {
+    await store.setRepresentation(resourceId, representation);
+    count += 1;
```

Comment:
Same thing here, `.generate` returns an iterator. This just also made me realize why some of these classes are perhaps not inhereting from AsyncHandler, since I think that would not be possible there.

Reply:
Ah good; perhaps comment that if this is the case indeed.